addthis_block/src/Plugin/Block/AddThisBlock.php

```diff
@@ -143,26 +143,54 @@ function blockForm($form, FormStateInterface $form_state) {
       '#title' => 'Display settings',
     );
 
-    // Retrieve settings.
-    $widget_type = $this->configuration['type'];
-    //$addthis_settings = AddThis::getInstance()->getBlockDisplaySettings();
-
 
-    // The list of formatters.
-    $formatter_options = AddThis::getInstance()->getDisplayTypes();
-
-    $form['settings']['addthis_settings']['type'] = array(
+    $form['settings']['addthis_settings']['share_services'] = array(
+      '#title' => t('Services'),
+      '#type' => 'textfield',
```

Comment:
@nerdstein  I want to get some more feedback on this since the interface for the checkboxes is so large. Providing a link to the options and textfield seems usable, but I agree that it is probably not the best approach. 

I want to keep mobile editing in mind here - a list of 300 checkboxes might provide more problems.